### PR TITLE
[Merged by Bors] - feat: port Algebra.Field.Defs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Tracking mathlib commit: [39af7d3bf61a98e928812dbc3e16f4ea8b795ca3](https://github.com/leanprover-community/mathlib/commit/39af7d3bf61a98e928812dbc3e16f4ea8b795ca3)

### CODE REVIEW

File: Mathlib/Algebra/Field/Defs.lean [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+universe u
+
+variable {α β K : Type _}
+
+/-- The default definition of the coercion `(↑(a : ℚ) : K)` for a division ring `K`
+is defined as `(a / b : K) = (a : K) * (b : K)⁻¹`.
+Use `coe` instead of `rat.cast_rec` for better definitional behaviour.
+-/
+def Rat.castRec [HasLiftT ℕ K] [HasLiftT ℤ K] [Mul K] [Inv K] : ℚ → K
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Field/Defs.lean:59]: I'm getting stuck on the lack of `HasLiftT` already - would appreciate a pointer to how this should look now

Comment (mcdoll) [reply Mathlib/Algebra/Field/Defs.lean:59]: It seems that we are lacking a port of https://github.com/leanprover-community/lean3port/blob/master/Leanbin/Init/Coe.lean - some parts of that file are obsolete though

Comment (semorrison) [reply Mathlib/Algebra/Field/Defs.lean]: I've ported this replacing `has_lift_t` with `CoeTail`. Unless someone knows a reason we really need to reproduce the `has_lift_t` distinction, this seems to work locally.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**